server/test/services/statuses/StatusServiceTest.java

```diff
@@ -0,0 +1,675 @@
+        .isEqualTo(expectedStatuses);
+  }
+
+  @Test
+  public void updateLocalizations_updatesExistingLocale() throws Exception {
+    ProgramModel program =
+        ProgramBuilder.newDraftProgram("English name", "English description")
+            .withLocalizedName(Locale.FRENCH, "existing French name")
+            .withLocalizedDescription(Locale.FRENCH, "existing French description")
+            .withLocalizedConfirmationMessage(Locale.FRENCH, "")
+            .setLocalizedSummaryImageDescription(
+                LocalizedStrings.of(
+                    Locale.US,
+                    "English image description",
+                    Locale.FRENCH,
+                    "existing French image description"))
+            .withStatusDefinitions(
+                new StatusDefinitions(ImmutableList.of(STATUS_WITH_EMAIL, STATUS_WITH_NO_EMAIL)))
+            .build();
+    assertThat(
+            applicationStatusesRepo
+                .lookupActiveStatusDefinitions(program.getProgramDefinition().adminName())
+                .getStatuses())
+        .isNotEmpty();
+
+    LocalizationUpdate updateData =
+        LocalizationUpdate.builder()
+            .setLocalizedDisplayName("new French name")
+            .setLocalizedDisplayDescription("new French description")
+            .setLocalizedSummaryImageDescription("new French image description")
+            .setLocalizedConfirmationMessage("")
+            .setStatuses(
+                ImmutableList.of(
+                    LocalizationUpdate.StatusUpdate.builder()
+                        .setStatusKeyToUpdate(STATUS_WITH_EMAIL_ENGLISH_NAME)
+                        .setLocalizedStatusText(
+                            Optional.of(STATUS_WITH_EMAIL_FRENCH_NAME + "-updated"))
+                        .setLocalizedEmailBody(
+                            Optional.of(STATUS_WITH_EMAIL_FRENCH_EMAIL + "-updated"))
+                        .build(),
+                    LocalizationUpdate.StatusUpdate.builder()
+                        .setStatusKeyToUpdate(STATUS_WITH_NO_EMAIL_ENGLISH_NAME)
+                        .setLocalizedStatusText(
+                            Optional.of(STATUS_WITH_NO_EMAIL_FRENCH_NAME + "-updated"))
+                        .build()))
+            .setScreens(ImmutableList.of())
+            .build();
+    ErrorAnd<ProgramDefinition, CiviFormError> result =
+        ps.updateLocalization(program.id, Locale.FRENCH, updateData);
+
+    assertThat(result.isError()).isFalse();
+    ProgramDefinition definition = result.getResult();
+    assertThat(definition.localizedName().get(Locale.FRENCH)).isEqualTo("new French name");
+    assertThat(definition.localizedDescription().get(Locale.FRENCH))
+        .isEqualTo("new French description");
+    assertThat(definition.localizedSummaryImageDescription().isPresent()).isTrue();
+    assertThat(definition.localizedSummaryImageDescription().get().get(Locale.FRENCH))
+        .isEqualTo("new French image description");
```

Comment:
for example, we don't care if this part of the localization worked correctly, that's already tested in the ProgramService tests.

Reply:
Moved it out to ProgramServiceTest